src/js/common/schemaform/ArrayField.jsx

```diff
@@ -0,0 +1,321 @@
+import React from 'react';
+import _ from 'lodash/fp';
+import Scroll from 'react-scroll';
+import { focusElement } from '../utils/helpers';
+
+import {
+  retrieveSchema,
+  toIdSchema,
+  getDefaultFormState,
+  deepEquals
+} from 'react-jsonschema-form/lib/utils';
+
+import { errorSchemaIsValid } from './helpers';
+
+const Element = Scroll.Element;
+const scroller = Scroll.scroller;
+
+const scrollToFirstError = () => {
+  setTimeout(() => {
+    const errorEl = document.querySelector('.usa-input-error, .input-error-date');
+    if (errorEl) {
+      const position = errorEl.getBoundingClientRect().top + document.body.scrollTop;
+      Scroll.animateScroll.scrollTo(position - 10, {
+        duration: 500,
+        delay: 0,
+        smooth: true
+      });
+      focusElement(errorEl);
+    }
+  }, 100);
+};
+
+/* Non-review growable table (array) field */
+export default class ArrayField extends React.Component {
+  constructor(props) {
+    super(props);
+
+    /*
+     * We always want to show one empty item in the array, so add that if necessary
+     *
+     * It will not get saved unless a user changes data in it
+     */
+    const formData = Array.isArray(props.formData) ? props.formData : null;
+    let items;
+    if (!formData || formData.length === 0) {
+      items = [getDefaultFormState(props.schema.items, null, props.registry.definitions) || {}];
+    } else {
+      items = getDefaultFormState(props.schema, formData, props.registry.definitions) || [];
+    }
+
+    /*
+     * We're keeping the editing state in local state because it's easier to manage and
+     * doesn't need to persist from page to page
+     */
+    this.state = {
+      items,
+      editing: items.map(() => false)
+    };
+
+    this.onItemChange = this.onItemChange.bind(this);
+    this.handleAdd = this.handleAdd.bind(this);
```

Comment:
I'm curious what you've been doing for this. I've normally done these bind calls for all methods that use `this` in their bodies (though I missed them here). But it's only really necessary if you're passing the methods as props to another component.

Do you always call `bind` or only when necessary?